### PR TITLE
resolve comments made first submission of ticket-4-GET-api-articles-by-article-id

### PR DESCRIPTION
**Pull request by samsonthompson**
resolved comments 
refactored some code to protect again SQL injection 
resolved testing paramaters to be specific

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -41,29 +41,26 @@ afterAll(() => db.end());
...
+                    article_id: 1,
+                    title: 'Living in the shadow of a great man',
+                    topic: 'mitch',
+                    author: 'butter_bridge',
+                    body: 'I find this existence challenging',
+                    created_at: '2020-07-09T20:11:00.000Z',
+                    votes: 100,
+                    article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+                }]
+                expect(actualArticle).toEqual(expectedArticle)
```
Comment (nc-jodie-cheung) [inline __tests__/app.test.js:63]: Using a more flexible jest matcher such as **toMatchObject or objectContaining** will make your tests more robust if the data was to change or extend in properties in the future. Then you can expect **title: expect.any(String** for example.

File: controllers/getArticles.controllers.js [hunk 2]
```diff
@@ -1,18 +1,11 @@
...
-               return Promise.reject({status: 404, message: 'Article not found'})
-            } else {
-                res.status(200).send({article : data})
-            }
-        })
+exports.getArticle = (req, res, next) => {
+    const id = req.params.article_id
+    fetchArticlesById(id)
+        .then((data) => {
+            res.status(200).send({article : data})
```
Comment (nc-jodie-cheung) [inline controllers/getArticles.controllers.js:7]: Nice that you are sending this back on a key of article 😄

File: models/getArticles.models.js [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+exports.fetchArticlesById = (id) => {
     return db.query(
         `SELECT * FROM articles
-        WHERE article_id = ${article_id}`)
+        WHERE article_id = $1`, [id])
     .then((article) => {
-        // console.log(result.rows, '<<< in the model return');
-        return article.rows
+        if (article.rows.length < 1){
+            return Promise.reject({status: 404, message: 'Article not found'})
```
Comment (nc-jodie-cheung) [inline models/getArticles.models.js:9]: Nice error handling here 👍

**Review by nc-jodie-cheung — COMMENTED** [review]
Nice work, please see the comments for this task 😄

File: __tests__/app.test.js (hunk 1 above)
Comment (samsonthompson) [reply __tests__/app.test.js:63]: resolved :)

**Pull request merged**